saagieapi/jobs/jobs.py

```diff
@@ -522,18 +556,24 @@ def run_with_callback(self, job_id: str, freq: int = 10, timeout: int = -1) -> D
         res = self.run(job_id)
         job_instance_id = res.get("runJob").get("id")
         final_status_list = ["SUCCEEDED", "FAILED", "KILLED"]
-        job_instance_info = self.get_instance(job_instance_id)
+        job_instance_info = self.get_instance(job_instance_id, pprint_result=False)
         state = job_instance_info.get("jobInstance").get("status")
         sec = 0
+
+        logging.info("⏳ Job id %s with instance %s has just been requested", job_id, job_instance_id)
         while state not in final_status_list:
-            to = False if timeout == -1 else sec >= timeout
-            if to:
-                raise TimeoutError("Last state known : " + state)
-            time.sleep(freq)
-            sec += freq
-            job_instance_info = self.get_instance(job_instance_id)
-            state = job_instance_info.get("jobInstance").get("status")
-            logging.info(f"Job id {job_id} with instance {job_instance_id} is currently : " + state)
+            with console.status(f"Job is currently {state}", refresh_per_second=100):
+                to = False if timeout == -1 else sec >= timeout
+                if to:
+                    raise TimeoutError("❌ Last state known : " + state)
```

Comment:
Should add fstring for l568, like:
raise TimeoutError(f"❌ Last state known : {state}")